frontend/src/components/GoalForm/index.js

```diff
@@ -587,10 +587,17 @@ export default function GoalForm({
 
       const updatedGoals = await createOrUpdateGoals(goals);
 
-      const updatedObjectives = updatedGoals && updatedGoals.length > 0
-        && updatedGoals[0] && updatedGoals[0].objectives && updatedGoals[0].objectives.length > 0
-        ? [...updatedGoals[0].objectives]
-        : [];
+      // this filter on updatedGoals will only ever return an array with a length of 1
+      // representing the goal being edited
+      // (only one goal can be edited at a time, and even multi grant goals
+      // are deduplicated on the backend)
+      const [goalForEditing] = updatedGoals.filter((goal) => {
```

Comment:
If we know that there is always going to be 1 value maybe a find updatedGoals[0].id ===.